### PR TITLE
Use DXSDK_DIR env variable instead of full path

### PR DESCRIPTION
**Pull request by TracerDS**
Use `DXSDK_DIR` env variable instead of providing full path to DX SDK dir;
That way MTA project wont be affected when `DXSDK_DIR` will be changed.

When creating projects via `premake5`, path to the `DXSDK_DIR` will be an absolute path. That might break if you decide to change location of the dir. You would have to change the location in the project settings too which might be annoying.

### CODE REVIEW

**Comment by patrikjuvonen** [comment]
Build is failing.

**Comment by TracerDS** [comment]
> Build is failing.

Thats probably because github cant detect `DXSDK_DIR` env variable, thus its failing to find these files.
But it fails only in win32 build...?

**Comment by patrikjuvonen** [comment]
> Thats probably because github cant detect `DXSDK_DIR` env variable, thus its failing to find these files. But it fails only in win32 build...?

Nothing has changed here though: https://github.com/multitheftauto/mtasa-blue/blob/master/.github/workflows/build.yaml#L42

It fails for win32 only because only win32 builds the client projects that make use of DX.

**Comment by TracerDS** [comment]
> > Thats probably because github cant detect `DXSDK_DIR` env variable, thus its failing to find these files. But it fails only in win32 build...?
> 
> Nothing has changed here though: https://github.com/multitheftauto/mtasa-blue/blob/master/.github/workflows/build.yaml#L42
> 
> It fails for win32 only because only win32 builds the client projects that make use of DX.

Cant reproduce it :/ It works perfectly for me but not for the github job

It looks like github has some [problems](https://github.com/multitheftauto/mtasa-blue/actions/runs/5037294023/jobs/9034029656#step:8:125) with env variables and dont recognize `DXSDK_DIR` variable

**Comment by botder** [comment]
I don't quiet understand what the point of this pull request is? The premake script is already using the environment variable and I don't think there is a requirement for an absolute path there.

**Comment by TracerDS** [comment]
> I don't quiet understand what the point of this pull request is? The premake script is already using the environment variable and I don't think there is a requirement for an absolute path there.

Absolute path is **currently** being provided to VS projects. The point of this PR was to provide an env variable to VS include path. You can always change the env variable, but changing absolute path in >6 projects is kinda... annoying.

**Comment by botder** [comment]
I still don't understand the issue. It works for me when I am using both absolute and relative path to DXFiles.

For example, this will generate projects each with a relative path (`..\utils\DXFiles`) to DXFiles.
```ps
$env:DXSDK_DIR = "utils/DXFiles"
.\win-create-projects.bat
```

**Comment by TracerDS** [comment]
> I still don't understand the issue. It works for me when I am using both absolute and relative path to DXFiles.
> 
> For example, this will generate projects each with a relative path (`..\utils\DXFiles`) to DXFiles.
> 
> ```postscript
> $env:DXSDK_DIR = "utils/DXFiles"
> .\win-create-projects.bat
> ```

Is your DX dir path in VS directories set to `$(DXSDK_DIR)` or to an absolute one?

**Comment by botder** [comment]
> Is your DX dir path in VS directories set to `$(DXSDK_DIR)` or to an absolute one?

It's relative.

image

**Comment by TracerDS** [comment]
> > Is your DX dir path in VS directories set to `$(DXSDK_DIR)` or to an absolute one?
> 
> It's relative.
> 
> image

Yeah, and my PR's purpose is to replace that path with `$(DXSDK_DIR)` env variable.
```xml
<AdditionalIncludeDirectories>..\vendor;$(DXSDK_DIR)\Include;..\vendor\...</AdditionalIncludeDirectories>
```

File: win-create-projects.bat [hunk 1]
```diff
@@ -9,6 +9,13 @@ utils\premake5.exe install_unifont
 rem Generate solutions
 utils\premake5.exe vs2022
 
+rem Check utils folder
+echo ------------------
+dir /B .\utils\
+echo ------------------
+dir /B .\utils\DXFiles
+echo ------------------
+
```
Comment (botder) [inline win-create-projects.bat]: This looks like debug code. It's not needed.
```suggestion
```

File: premake5.lua [hunk 2]
```diff
@@ -45,7 +45,9 @@ workspace "MTASA"
 	symbols "On"
 	flags "MultiProcessorCompile"
 
-	dxdir = os.getenv("DXSDK_DIR") or ""
+	dxsdk = os.getenv("DXSDK_DIR") and "$(DXSDK_DIR)" or ""
+	print(('DXSDK: [%s] [%s]'):format(dxsdk, os.getenv("DXSDK_DIR")))
+
 	includedirs {
```
Comment (botder) [inline premake5.lua]:
```suggestion
	includedirs {
```

File: premake5.lua [hunk 3]
```diff
@@ -110,10 +112,10 @@ workspace "MTASA"
 		defines { "WIN32", "_WIN32", "_WIN32_WINNT=0x601", "_MSC_PLATFORM_TOOLSET=$(PlatformToolsetVersion)" }
 		buildoptions { "/Zc:__cplusplus" }
 		includedirs {
-			path.join(dxdir, "Include")
+			dxsdk.."Include"
 		}
 		libdirs {
-			path.join(dxdir, "Lib/x86")
+			dxsdk.."Lib/x86"
 		}
```
Comment (botder) [inline premake5.lua]: This is enough:
```lua
		includedirs { "$(DXSDK_DIR)/Include" }
		libdirs { "$(DXSDK_DIR)/Lib/x86" }
```

**Review by botder — CHANGES_REQUESTED** [review]
Can you show a warning if the environment variable `DXSDK_DIR` is empty or unset (only on Windows)?

File: win-create-projects.bat (hunk 1 above)
Comment (TracerDS) [reply win-create-projects.bat]: > This looks like debug code. It's not needed.

Yeah, I tried to pinpoint the culprit. Couldnt do it without commiting ;/

**Comment by botder** [comment]
You should also change [build.yaml](https://github.com/multitheftauto/mtasa-blue/blob/master/.github/workflows/build.yaml#L42) and try removing the `./` prefix there

**Comment by botder** [comment]
There is no reason to change the current code. It also works with a relative path.

**Pull request closed without merging**